src/Icicle/Repl.hs

```diff
@@ -19,120 +20,74 @@ module Icicle.Repl (
+import qualified Icicle.Source.Type               as ST
+import qualified Icicle.Storage.Dictionary.TextV1 as DictionaryText
+import qualified Icicle.Storage.Dictionary.Toml   as DictionaryToml
 
 import           P
 
 import           Control.Monad.Trans.Class
 import           Control.Monad.Trans.Either
-import         X.Control.Monad.Trans.Either
+import           X.Control.Monad.Trans.Either
 
 import           Data.Either.Combinators
-import           Data.Functor.Identity
-import           Data.Text                          (Text)
-import qualified Data.Text                          as T
-import qualified Data.Text.IO                         as T
-import qualified Data.Traversable                   as TR
-import qualified Data.Map                           as M
+import qualified Data.Map                         as M
+import           Data.Text                        (Text)
+import qualified Data.Text                        as T
+import qualified Data.Text.IO                     as T
+import qualified Data.Traversable                 as TR
 
 import           System.IO
 
-import qualified Text.ParserCombinators.Parsec      as Parsec
+import qualified Text.ParserCombinators.Parsec    as Parsec
 
 data ReplError
- = ReplErrorParse   Parsec.ParseError
- | ReplErrorDesugar (STD.DesugarError Var)
- | ReplErrorCheck   (SC.CheckError Parsec.SourcePos Var)
- | ReplErrorConvert (STC.ConvertError Parsec.SourcePos Var)
- | ReplErrorDecode  S.ParseError
- | ReplErrorRuntime (S.SimulateError ())
- | ReplErrorFlatten (AS.FlattenError () Text)
- | ReplErrorProgram (AC.ProgramError () Text APF.Prim)
+ = ReplErrorCompile        P.CompileError
+ | ReplErrorRuntime        (S.SimulateError ())
  | ReplErrorDictionaryLoad DictionaryToml.DictionaryImportError
+ | ReplErrorDecode         S.ParseError
  deriving (Show)
 
 annotOfError :: ReplError -> Maybe Parsec.SourcePos
 annotOfError e
  = case e of
-    ReplErrorParse sp
-     -> Just
-      $ Parsec.errorPos sp
-    ReplErrorDesugar _
-     -> Nothing
-    ReplErrorCheck       e'
-     -> SC.annotOfError  e'
-    ReplErrorConvert     e'
-     -> STC.annotOfError e'
-    ReplErrorDecode  _
-     -> Nothing
+    ReplErrorCompile d
+     -> P.annotOfError d
     ReplErrorRuntime _
      -> Nothing
-    ReplErrorFlatten _
-     -> Nothing
-    ReplErrorProgram _
-     -> Nothing
     ReplErrorDictionaryLoad _
      -> Nothing
+    ReplErrorDecode  _
+     -> Nothing
 
 instance Pretty ReplError where
  pretty e
   = case e of
-     ReplErrorParse p
-      -> "Parse error:" <> line
-      <> indent 2 (text $ show p)
-     ReplErrorDesugar d
-      -> "Desugar error:" <> line
-      <> indent 2 (text $ show d)
-     ReplErrorCheck ce
-      -> "Check error:" <> line
-      <> indent 2 (pretty ce)
-     ReplErrorConvert ce
-      -> "Convert error:" <> line
-      <> indent 2 (pretty ce)
-     ReplErrorDecode d
-      -> "Decode error:" <> line
+     ReplErrorCompile d
+      -> "Compile error:" <> line
```

Comment:
Probably don't need this extra bit of text and indent. Plus
```
Compile error:
  ParseError:
    blahdiblah
```
is less clear that just `ParseError`